### PR TITLE
WIP: Add feature item owner should be able to edit item

### PR DESCRIPTION
**Pull request by nomthysmylz**
Work in Progress:

Addresses #11 
- Added methods `doUpdateItem` and `updateItemForm` to `ItemController`
- Added web-gateway updateitem form
- Add button "Update Item" to "view item" view if user is owner

### TODO  
- Invoke service call in doUpdateItem
- Implement updateItem service call in item-api
- Add updateItem test

### CODE REVIEW

File: web-gateway/app/controllers/ItemController.scala [hunk 1]
```diff
@@ -129,6 +129,39 @@ class ItemController(messagesApi: MessagesApi, userService: UserService, itemSer
     }
   }
 
+  def updateItemForm(itemId: UUID) = Action.async{ implicit rh =>
+    requireUser(user => loadNav(user).flatMap { implicit nav =>
+      val itemFuture = itemService.getItem(itemId)
+        .handleRequestHeader(authenticate(user)).invoke()
+      for( item <- itemFuture ) yield{
+        val (duration, durationUnits) = (item.auctionDuration.getSeconds match{
```
Comment (ignasi35) [inline web-gateway/app/controllers/ItemController.scala]: You don't need a for comprehension here. You can directly map over the future returned by `invoke()`  (which is what the for comprehension is doing under the covers anyway).

File: web-gateway/app/controllers/ItemController.scala [hunk 2]
```diff
@@ -129,6 +129,39 @@ class ItemController(messagesApi: MessagesApi, userService: UserService, itemSer
...
+          case l: Long => if(l < 60)
+              Seq((l, ChronoUnit.SECONDS))
+            else
+              ChronoUnit.values
+              .dropWhile(_ != ChronoUnit.MINUTES)
+              .takeWhile(_ != ChronoUnit.MONTHS)
+              .map{ unit => (l/unit.getDuration.getSeconds, unit) }.toSeq
+        }).filterNot(_._1 == 0).reverse.head
```
Comment (ignasi35) [inline web-gateway/app/controllers/ItemController.scala]: Consider moving this conversion to a separate function.

File: web-gateway/app/views/item.scala.html [hunk 3]
```diff
@@ -93,11 +93,21 @@
...
+                </form>
+            </div>
+            }
+            @if(! Seq(ItemStatus.Completed, ItemStatus.Cancelled).contains(item.status)) {
+            <div class="columns medium-6" >
+                <a href="@routes.ItemController.updateItemForm(item.safeId)" class="button primary">@Messages("updateItem")</a>
+            </div>
+            }
+        </div>
     }
```
Comment (ignasi35) [inline web-gateway/app/views/item.scala.html]: the edit page is a bit more tricky. I understand that it's WIP as is is now

Remember that when the auction already started only the description may be edited.

Comment (nomthysmylz) [inline web-gateway/app/views/item.scala.html]: will update this

File: item-impl/src/main/scala/com/example/auction/item/impl/ItemServiceImpl.scala [hunk 4]
```diff
@@ -31,6 +31,14 @@ class ItemServiceImpl(registry: PersistentEntityRegistry, itemRepository: ItemRe
     }
   })
 
+  override def updateItem(id: UUID) = authenticated(userId => ServerServiceCall{ item =>
```
Comment (thhofer) [inline item-impl/src/main/scala/com/example/auction/item/impl/ItemServiceImpl.scala]: Am I missing something, or is the verification that the caller is indeed the creator of the item only performed in the view?

Comment (ignasi35) [inline item-impl/src/main/scala/com/example/auction/item/impl/ItemServiceImpl.scala]: @thhofer IIUC this is WIP and @nomthysmylz still hasn't completed that bit ;-)

**Comment by TimMoore** [comment]
Hi @nomthysmylz are you still working on this?

**Comment by TimMoore** [comment]
I'll close this for now, but please let us know if you plan to pick this back up and we can reopen.

**Pull request closed without merging**